x-pack/plugins/apm/server/routes/alerts/route.ts

```diff
@@ -40,12 +41,16 @@ const transactionErrorRateChartPreview = createApmServerRoute({
   handler: async (
     resources
   ): Promise<{ errorRateChartPreview: Array<{ x: number; y: number }> }> => {
-    const setup = await setupRequest(resources);
+    const [setup, apmEventClient] = await Promise.all([
+      setupRequest(resources),
+      getApmEventClient(resources),
+    ]);
     const { params } = resources;
     const { _inspect, ...alertParams } = params.query;
 
     const errorRateChartPreview = await getTransactionErrorRateChartPreview({
-      setup,
+      config: setup.config,
+      apmEventClient,
```

Comment:
Can't you do:

```suggestion
      config: resources.config,
      apmEventClient,
```

And this way we don't need setup anymore, right?